pkg/basecluster/preparegit_test.go

```diff
@@ -0,0 +1,48 @@
+package basecluster
+
+import (
+	"errors"
+	"fmt"
+	"github.com/arlonproj/arlon/pkg/argocd"
+	"github.com/arlonproj/arlon/pkg/gitutils"
+	gogit "github.com/go-git/go-git/v5"
+	"github.com/otiai10/copy"
+	"gotest.tools/assert"
+	"os"
+	"path"
+	"testing"
+)
+
+func TestGitPreparation(t *testing.T) {
+	subdirName := "requires_prep"
+	repoRevision := "master"
+	_, srcGitDir := createFileSystemBasedRepo(t, subdirName)
+	fmt.Println(srcGitDir)
+	creds := &argocd.RepoCreds{}
+	repoUrl := "file://" + srcGitDir
+	_, err := ValidateGitDir(creds, repoUrl, repoRevision, subdirName)
+	assert.Assert(t, errors.Is(err, ErrNoKustomizationYaml), "unexpected validation error: %s", err)
+	fmt.Println("got expected error:", err)
+	clustName, changed, err := PrepareGitDir(creds, repoUrl, repoRevision, subdirName)
+	assert.NilError(t, err, "failed to prepare git directory")
+	assert.Assert(t, changed, "git dir preparation resulted in no changes")
+	assert.Equal(t, clustName, "capi-quickstart", "unexpected cluster name: %s", clustName)
+	_, err = ValidateGitDir(creds, repoUrl, repoRevision, subdirName)
+	assert.NilError(t, err, "unexpected 2nd validation error: %s", err)
+}
+
+func createFileSystemBasedRepo(t *testing.T, subdirName string) (*gogit.Repository, string) {
+	srcDir := path.Join("testdata", subdirName)
+	tmpDir, err := os.MkdirTemp("", "arlon-unittest-")
```

Comment:
A small nitpick. We're not cleaning up the tmp dir we create.
Would be nice to add something like so:
```go
	t.Cleanup(func() {
		os.RemoveAll(tmpDir)
	})
```

Reply:
done